tests/cases/fourslash/quickInfoInOptionalChain.ts

```diff
@@ -0,0 +1,28 @@
+/// <reference path='fourslash.ts'/>
+//
+// @strict: true
+//
+//// interface A {
+////   arr: string[];
+//// }
+////
+//// function test(a?: A): string {
+////   return a?.ar/*1*/r.length ? "A" : "B";
+//// }
+//// 
+//// interface Foo { bar: { baz: string } };
+//// declare const foo: Foo | undefined;
+////
+//// if (foo?.b/*2*/ar.b/*3*/az) {}
+////
+//// interface Foo2 { bar?: { baz: { qwe: string } } };
+//// declare const foo2: Foo2;
+////
+//// if (foo2.b/*4*/ar?.b/*5*/az.q/*6*/we) {}
+
+verify.quickInfoAt("1", "(property) A.arr: string[]");
+verify.quickInfoAt("2", "(property) Foo.bar: {\n    baz: string;\n}");
+verify.quickInfoAt("3", "(property) baz: string | undefined");
```

Comment:
I'm confused as to the end result here; both `bar` and `baz` themselves are not declared as potentially undefined, and yet `bar` is not `| undefined` while `baz` is. Why should they behave differently?

Reply:
I admit this is a little bit confusing and I'm open for feedback how to improve this further. 

`baz` has `| undefined` because it's the end of the optional chain expression and the whole optional chain expression has `| undefined`. Without this bit you'd look at an if statement that would always~ be truthy. That `| undefined` is here to inform you that the whole expression can indeed result in both branches to be taken.

OTOH, `bar` doesn't have it because with it it looks really weird that we can just chain off this expression (`foo?.bar`) without guarding the `.baz` property access

Reply:
I guess that makes sense, though I'd probably personally defer to @DanielRosenwasser 😄

Reply:
That's a good catch. I think that it's worth unifying those - I'd expect `x?.method;` to include `undefined` based on the given described intuition. I see how this is quite subtle though so perhaps not everybody would agree with me on this one and that's fine.

I think that this find doesn't have to block this PR, I can just prepare a follow up fixing it at some later point in time. Up to you though.